homeassistant/components/keenetic_ndms2/router.py

```diff
@@ -0,0 +1,197 @@
+        """Return the model of the hub."""
+        return self._router_info.model if self._router_info else None
+
+    @property
+    def firmware(self):
+        """Return the firmware of the hub."""
+        return self._router_info.fw_version if self._router_info else None
+
+    @property
+    def manufacturer(self):
+        """Return the firmware of the hub."""
+        return self._router_info.manufacturer if self._router_info else None
+
+    @property
+    def available(self):
+        """Return if the hub is connected."""
+        return self._available
+
+    @property
+    def consider_home_interval(self):
+        """Config entry option defining number of seconds from last seen to away."""
+        return timedelta(seconds=self.config_entry.options[CONF_CONSIDER_HOME])
+
+    @property
+    def signal_update(self):
+        """Event specific per router entry to signal updates."""
+        return f"keenetic-update-{self.host}"
+
+    async def async_add_defaults(self):
+        """Populate default options."""
+        data = dict(self.config_entry.data)
+        options = {
+            CONF_SCAN_INTERVAL: DEFAULT_SCAN_INTERVAL,
+            CONF_CONSIDER_HOME: DEFAULT_CONSIDER_HOME,
+            CONF_INTERFACE: DEFAULT_INTERFACE,
+            CONF_TRY_HOTSPOT: True,
+            CONF_INCLUDE_ARP: True,
+            CONF_INCLUDE_ASSOCIATED: True,
+            **self.config_entry.options,
+        }
+
+        if options.keys() - self.config_entry.options.keys():
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, data=data, options=options
+            )
+
+    async def request_update(self):
+        """Request an update."""
+        if self.progress is not None:
+            await self.progress
+            return
+
+        self.progress = self.hass.async_create_task(self.async_update())
+        await self.progress
+
+        self.progress = None
+
+    async def async_update(self):
+        """Update devices information."""
+        await self.hass.async_add_executor_job(self._update_devices)
+        async_dispatcher_send(self.hass, self.signal_update)
+
+    async def async_setup(self):
+        """Set up the connection."""
+        self._client = Client(
+            TelnetConnection(
+                self.config_entry.data[CONF_HOST],
+                self.config_entry.data[CONF_PORT],
+                self.config_entry.data[CONF_USERNAME],
+                self.config_entry.data[CONF_PASSWORD],
+            )
+        )
+
+        try:
+            await self.hass.async_add_executor_job(self._update_router_info)
+        except ConnectionException as error:
+            raise ConfigEntryNotReady from error
+
+        await self.async_add_defaults()
+        await self.request_update()
+
+        @callback
+        async def periodic_update(_now):
+            await self.request_update()
+            self._cancel_periodic_update = async_track_point_in_utc_time(
```

Comment:
Look into our update coordinator. I think we can use that instead.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
I was trying that out. Faced two issues:
* I can't change polling interval with integration options change
* when I was trying to subscribe from KeeneticTracker(ScannerEntity) I was facing an issue with that at the first scan we could get an empty list of connected devices and nobody will subscribe to the coordinator.

Currently Router is a single point doing the polling for a particular router and does not benefit from coordinator

Reply:
We can use this helper instead:
https://github.com/home-assistant/core/blob/82607977efaa385e6097509ada5a1153f95214b2/homeassistant/helpers/event.py#L1175-L1180